src/main.ts

```diff
@@ -0,0 +1,146 @@
+                account = parsedSchema.sender.login;
+                return {
+                    message: `**New/updated GitHub issue**: ${parsedSchema.issue.title}`,
+                    url: parsedSchema.issue.html_url,
+                    details: `Issue state: ${parsedSchema.issue.state}  - assignee: ${parsedSchema.issue.assignee}`
+                };
+            case EventType.ISSUE_COMMENT:
+                parsedSchema = schemaOnIssueComment.validateSync(eventPayloadText);
+                return {
+                    message: `**A Github issue comment was posted** by ${account} with content ${parsedSchema.comment.body}`,
+                    url: parsedSchema.issue.html_url,
+                    details: `Issue state: ${parsedSchema.issue.state}  - assignee: ${parsedSchema.issue.assignee}`
+                };
+            default:
+                return {
+                    message: eventName ? eventName.replace(/_/g, '.') : 'A GitHub Actions event has occurred',
+                    url: `https://github.com/${repository}/actions`,
+                    details: core.getInput('details')
+                };
+        }
+    } catch (error) {
+        let errorDetails;
+        if (error instanceof ValidationError) {
+            errorDetails = error.errors;
+        } else {
+            errorDetails = error.message;
+        }
+        return core.setFailed(`ERROR : ${errorDetails}`);
+    }
+};
+
+
+async function notifyTeams(notificationMessage: NotificationMessage) {
+    if (!(core.getInput('webhook_url'))) {
+        core.setFailed('Error : Webhook URL configuration is missing. Aborting');
+        process.exit(1);
+    }
+    const matches = core.getInput('webhook_url').match(/^https?:\/\/([^/?#]+)(.*)/i);
+    const hostnameMatch = matches && matches[1];
+    const pathMatch = matches && matches[2];
+
+    let requestBodyData = JSON.stringify(notificationTemplate);
+    requestBodyData = requestBodyData
+        .replace(/GITHUB_WORKFLOW/g, `${workflow}`)
+        .replace(/GITHUB_REPOSITORY/g, `${repository}`)
+        .replace(/GITHUB_REF/g, `${branch}`)
+        .replace(/GITHUB_TRIGGER_EVENT_DETAILS/g, `${notificationMessage.details}`)
+        .replace(/GITHUB_TRIGGER_EVENT/g, `${notificationMessage.message}`)
+        .replace(/GITHUB_EVENT_URL/g, `${notificationMessage.url}`)
+        .replace(/GITHUB_STATUS/g, `${core.getInput('job_status')}`);
+
+    const options = {
+        hostname: `${hostnameMatch}`,
+        port: 443,
+        path: `${pathMatch}`,
+        method: 'POST',
+        headers: {
+            'Content-Type': 'application/json',
+            'Content-Length': requestBodyData.length
+        }
+    };
+
+    const req = https.request(options, (res) => {
+        if (res.statusCode >= 400) {
+            core.setFailed(`Action failed with status code  ${res.statusCode}`);
+        } else {
+            core.info(`statusCode: ${res.statusCode}`);
+        }
+
+        res.on('data', (d) => {
```

Comment:
the 'data' event can possibly only give you partial data of the response (a.k.a. a 'chunk') - which means there may be more than one 'data' event. technically we need to aggregate all the chunks until we get the 'end' event.
it can get complicated. that's why I highly recommend not to use node's native http library, and use something like 'node-fetch' instead.

https://stackoverflow.com/a/19539521/1365324

Reply:
Convinced! I gave node-fetch a try, please take a look and let me know if this is what you were going for
```
    fetch(core.getInput('webhook_url'), {
        method: 'POST',
        body: requestBodyData,
        headers: {
            'Content-Type': 'application/json',
            'Content-Length': requestBodyData.length
        }
    }).then((res) => {
        if (!res) {
            core.setFailed('No response from action');
        } else if (res.status && res.status >= 400) {
            core.setFailed(`Action failed with status code  ${res.status}`);
        } else {
            core.info(`statusCode: ${res.status}`);
        }
    }).catch((err) => {
        core.error(err.message);
        core.setFailed(`Action failed with error ${err.message}`);
    });
```

Reply:
> Convinced! I gave node-fetch a try, please take a look and let me know if this is what you were going for
> 
> ```
>     fetch(core.getInput('webhook_url'), {
>         method: 'POST',
>         body: requestBodyData,
>         headers: {
>             'Content-Type': 'application/json',
>             'Content-Length': requestBodyData.length
>         }
>     }).then((res) => {
>         if (!res) {
>             core.setFailed('No response from action');
>         } else if (res.status && res.status >= 400) {
>             core.setFailed(`Action failed with status code  ${res.status}`);
>         } else {
>             core.info(`statusCode: ${res.status}`);
>         }
>     }).catch((err) => {
>         core.error(err.message);
>         core.setFailed(`Action failed with error ${err.message}`);
>     });
> ```
rather than `if (res.status && res.status >= 400)` I would suggest `if (!res.ok)`

Reply:
done